common/src/main/kotlin/Locale.kt

```diff
@@ -0,0 +1,245 @@
+
+        /**
+         * Italian.
+         */
+        val ITALIAN = Locale("it")
+
+        /**
+         * Japanese.
+         */
+        val JAPENESE = Locale("ja")
+
+        /**
+         * Korean.
+         */
+        val KOREAN = Locale("ko")
+
+        /**
+         * Lithuanian.
+         */
+        val LITHUANIAN = Locale("lt")
+
+        /**
+         * Norwegian.
+         */
+        val NORWEGIAN = Locale("no")
+
+        /**
+         * Polish.
+         */
+        val POLISH = Locale("pl")
+
+        /**
+         * Portuguese (Brazil).
+         */
+        val PORTUGUESE_BRAZIL = Locale("pt", "BR")
+
+        /**
+         * Romanian.
+         */
+        val ROMANIAN = Locale("ro")
+
+        /**
+         * Russian.
+         */
+        val RUSSIAN = Locale("ru")
+
+        /**
+         * Spanish (Spain).
+         */
+        val SPANISH_SPAIN = Locale("es", "ES")
+
+        /**
+         * Swedish.
+         */
+        val SWEDISH = Locale("sv", "SE")
+
+        /**
+         * Thai.
+         */
+        val THAI = Locale("th")
+
+        /**
+         * Turkish.
+         */
+        val TURKISH = Locale("tr")
+
+        /**
+         * Ukrainian.
+         */
+        val UKRAINIAN = Locale("uk")
+
+        /**
+         * Vietnamese.
+         */
+        val VIETNAMESE = Locale("vi")
+
+
+        /**
+         * All languages [supported by Discord](https://discord.com/developers/docs/reference#Locales).
+         */
+        val ALL = listOf(
+            ENGLISH_UNITED_STATES, ENGLISH_GREAT_BRITAN,
+            BULGARIAN,
+            CHINESE_CHINA, CHINESE_TAIWAN,
+            CROATIAN,
+            CZECH,
+            DANISH,
+            DUTCH,
+            FINNISH,
+            FRENCH,
+            GERMAN,
+            GREEK,
+            HINDI,
+            HUNGARIAN,
+            ITALIAN,
+            JAPENESE,
+            KOREAN,
+            LITHUANIAN,
+            NORWEGIAN,
+            POLISH,
+            PORTUGUESE_BRAZIL,
+            ROMANIAN,
+            RUSSIAN,
+            SPANISH_SPAIN,
+            SWEDISH,
+            THAI,
+            TURKISH,
+            UKRAINIAN,
+            VIETNAMESE
+        )
+
+        // We accept both "_" and "-" as a separator, because Discord doesn't really document it
+        // https://regex101.com/r/fVPdR8/2
+        private val languageTagFormat = "(\\w{2})(?:[_\\-](\\w{2}))?".toRegex()
+
+        /**
+         * Decodes the language from a `languageCode_countryCode` or `languageCode` format.
+         */
+        fun fromString(string: String): Locale {
+            val match = languageTagFormat.matchEntire(string) ?: error("$string is not a valid Locale")
+            val (language) = match.destructured
+            val country = match.groupValues[2]
+
+            return ALL.firstOrNull { (l, c) ->
+                language == l && country == (c ?: "")
+            } ?: error("This locale is not supported by Discord")
```

Comment:
Should this really throw an exception? There is still a public constructor.

Reply:
So you mean I should just default to Locale(country, language)?

Reply:
Yes, this way Discord randomly adding new locales won't cause exceptions.

Reply:
This is a common pattern in our `types` values, making the system error tolerant when Discord adds something to this.